### PR TITLE
Adding GitLab pipeline.

### PR DESCRIPTION
**Pull request by getvictor**
For #17181

Demo video: https://www.loom.com/share/df887c7f64d746608cea4ac31225b4d4

### CODE REVIEW

**Comment by getvictor** [comment]
@mostlikelee are you seeing this in Toast?

**Review by mostlikelee — APPROVED** [review]
LGTM

**Review by noahtalerman — APPROVED** [review]

**Comment by noahtalerman** [comment]
@getvictor I didn't review changes. Approving to unblock y'all.

Also heads up that I made you CODEOWNER of this repo to speed things up: 
- #49

**Pull request merged**
